### PR TITLE
Selectable User-Agent for HTTP requests

### PR DESCRIPTION
**Pull request by oddluck**
Closes https://github.com/ProgVal/Limnoria/issues/1382

### CODE REVIEW

**Comment by ProgVal** [comment]
The way it's implemented, changes to the config var won't take affect until the bot is restarted.

And you can merge the two `try` blocks

File: src/conf.py [hunk 1]
```diff
@@ -1330,13 +1331,17 @@ def defaultHttpHeaders(network, channel):
     try:
         language = supybot.protocols.http.requestLanguage.getSpecific(
                 network, channel)()
+        agent = random.choice(supybot.protocols.http.userAgent.getSpecific(
+                network, channel)())
     except registry.NonExistentRegistryEntry:
-        pass # Starting up; language will be set by HttpRequestLanguage later
+        pass
```
Comment (ProgVal) [inline src/conf.py:1337]: Leave the comment, it's unclear why `NonExistentRegistryEntry` may be raised otherwise

File: src/conf.py [hunk 2]
```diff
@@ -1346,12 +1351,23 @@ def setValue(self, v):
         super(HttpRequestLanguage, self).setValue(v)
         utils.web.defaultHeaders = defaultHttpHeaders(None, None)
 
+class HttpUserAgent(registry.CommaSeparatedListOfStrings):
```
Comment (ProgVal) [inline src/conf.py:1354]: `HttpUserAgents` (plural)

File: src/conf.py [hunk 3]
```diff
@@ -1346,12 +1351,23 @@ def setValue(self, v):
...
+
 registerChannelValue(supybot.protocols.http, 'requestLanguage',
     HttpRequestLanguage('', _("""If set, the Accept-Language HTTP header will be set to this
     value for requests. Useful for overriding the auto-detected language based on
     the server's location.""")))
 
 
+registerChannelValue(supybot.protocols.http, 'userAgent',
+    HttpUserAgent([], _("""If set, the User-Agent HTTP header
+    will be set to a randomly selected value from this list for requests.""")))
```
Comment (ProgVal) [inline src/conf.py:1369]: Should mention it's a comma-separated list.

File: src/conf.py [hunk 4]
```diff
@@ -1351,8 +1351,8 @@ def setValue(self, v):
         super(HttpRequestLanguage, self).setValue(v)
         utils.web.defaultHeaders = defaultHttpHeaders(None, None)
 
-class HttpUserAgent(registry.CommaSeparatedListOfStrings):
-    """Must be a valid HTTP User-Agent value."""
+class HttpUserAgents(registry.CommaSeparatedListOfStrings):
+    """Must be a valid HTTP Accept-Language value."""
```
Comment (ProgVal) [inline src/conf.py:1355]: woops!

Comment (oddluck) [reply src/conf.py]: Sorry about that.

**Pull request merged**